docs: rewrite Actor-to-Actor integration page (#2546)
apify/apify-docs: sources/platform/integrations/actors/integration_ready_actors.md

:::

Any Actor can be used as an integration target, but Actors designed for the role give users a smoother experience. The rest of this page assumes you are developing such an Actor.

## General guidelines

If your Actor is supposed to be used as an integration, it will most likely have an input that can be described as two groups of fields. The first group is the "static" part of the input - the fields that have the same value whenever the integration is triggered. The second, "dynamic", group are fields that are specific to the triggering event - information from the run or build that triggered the integration.

The Actor should hide its complexity from users and take all the "dynamic" fields from the implicit `payload` field - it is attached automatically. This way, users don't have to take care of passing in variables on their own and only need to take care of the static part of the input.

Only the dataset ID is passed to the Actor as input, not the dataset contents. The Actor needs to fetch the contents of the dataset itself. Ideally, it should not load the full dataset at once, as it might be too large to fit in memory, but rather process it in batches.

## Example

> To illustrate the above, here is a simplified example of an Actor that uploads a dataset to a table/collection in some database.

Reviewer: ```suggestion
To illustrate the above, here is a simplified example of an Actor that uploads a dataset to a database table or collection.
```

Start with an input that looks something like this:

- `datasetId: string` - ID of the dataset to upload
- `connectionString: string` - credentials for the database connection
- `tableName: string` - name of the table or collection

> 

Reviewer: here you have lowercase, L67-68 uppercase. Lets pick one and be consistent

Explicitly stating what is the expected input when Actor is being used as an integration is a preferred way.

However, if the Actor is only supposed to be used as an integration, use a different input schema:

- `connectionString: string` - Credentials for the database connection

> - `tableName: string` - Name of table / collection

Reviewer: no need to use symbol when a simple word does the same job 

```suggestion
- `tableName: string` - Name of table or collection
```
